l10n_nl_bank/scripts/generate_bank_data.py

```diff
@@ -0,0 +1,175 @@
+    """
+    Converts a string to title case, while keeping specific abbreviations
+    like 'N.V.', 'B.V.', and 'NV' in full caps.
+    """
+    title_cased = text.title()
+    fix_map = {
+        "Nv": "NV",
+        "Bv": "BV",
+        "Ing": "ING",
+        "Asn": "ASN",
+        "Abn": "ABN",
+        "Blg": "BLG",
+        "Bnp": "BNP",
+        "Nibc": "NIBC",
+        "Xe": "XE",
+        "Bunq": "BUNQ",
+        "Hsbc": "HSBC",
+        "Kbc": "KBC",
+        "Keb": "KEB",
+        "Mufg": "MUFG",
+    }
+    words = title_cased.split()
+    corrected_words = [fix_map.get(word, word) for word in words]
+    return " ".join(corrected_words)
+
+
+def fetch_and_process_data():
+    """
+    Fetches the bank data from the Excel file, finds the header row dynamically,
+    and processes the data.
+    """
+    logging.info("📥 Fetching and processing bank data from %s...", DATA_URL)
+    try:
+        # Step 1: Read the Excel file without assuming a header to find its location.
+        df_no_header = pd.read_excel(DATA_URL, engine="openpyxl", header=None)
+
+        header_row_index = -1
+        bic_col_name = "BIC"
+        name_col_name = "Naam betaaldienstverlener"
+
+        # Step 2: Search for the header row within the first 10 rows of the file.
+        for i, row in df_no_header.head(10).iterrows():
+            row_values = [str(v).strip() for v in row.values]
+            if bic_col_name in row_values and name_col_name in row_values:
+                header_row_index = i
+                break
+
+        if header_row_index == -1:
+            logging.error(
+                "❌ Error: Could not find the header row containing '%s' and '%s'.",
+                bic_col_name,
+                name_col_name,
+            )
+            return []
+
+        logging.info(
+            "✅ Found headers on row %s. Proceeding with data extraction.",
+            header_row_index + 1,
+        )
+
+        # Step 3: Re-read the Excel file,
+        # telling pandas which row is the correct header.
+        df = pd.read_excel(DATA_URL, engine="openpyxl", header=header_row_index)
+
+        processed_banks = []
+        used_ids = set()
+
+        for _, row in df.iterrows():
+            # Use the known correct column names
+            bic = str(row[bic_col_name]).strip()
+            name_raw = str(row[name_col_name]).strip()
+
+            if not bic or not name_raw or bic == "nan" or name_raw == "nan":
+                continue
+
+            name = to_custom_title_case(name_raw)
+            base_id = bic[:4].upper().replace("_", "")
+            record_id = base_id
+
+            counter = 2
+            while record_id in used_ids:
+                record_id = f"{base_id}_{counter}"
+                counter += 1
+
+            used_ids.add(record_id)
+            processed_banks.append((record_id, name, bic))
+
+        logging.info("✅ Data processed successfully.")
+        return sorted(processed_banks, key=lambda x: x[1])
```

Comment:
```suggestion
        return sorted(processed_banks, key=lambda x: x[0])
```

to keep sorting by bic as the existing file does, should give a smaller diff and keep ordering stable for name changes